src/pages/Profile/components/Friends/index.tsx

```diff
@@ -4,19 +4,86 @@
  *         a popup appears displaying that friends featured deck.
  */
 
-import React from "react";
+import React, { useEffect, useState } from "react";
 import "./friends.css";
-import { useFriendsList } from "pages/Profile/hooks/useFriendsList";
 import FriendDeckPopup from "pages/Profile/components/FriendDeckPopup";
-import { IUser } from "combatcritters-ts";
+import { ICard, IDeck, IUser } from "combatcritters-ts";
+import Popup from "components/Popup";
+import { ClientSingleton } from "ClientSingleton";
+import { toast } from "react-toastify";
 
 interface FriendsProps {
   friends: IUser[];
   setFriends: (friends: IUser[]) => void;
 }
 const Friends: React.FC<FriendsProps> = ({ friends, setFriends }) => {
-  const { selectedFriend, setSelectedFriend, showDeck, setShowDeck, onFriendClick } =
-    useFriendsList(friends, setFriends);
+  const [selectedFriend, setSelectedFriend] = useState<IUser | null>(null);
+  const [showDeck, setShowDeck] = useState(false);
+  const [featuredDeck, setFeaturedDeck] = useState<IDeck | null>(null);
+  const [featuredDeckCards, setFeaturedDeckCards] = useState<ICard[] | null>(
+    null
+  );
+
+  const onFriendClick = (user: IUser) => {
+    setSelectedFriend(user);
+  };
+
+  /**
+   * On mount, fetch the user's friends
+   */
+  useEffect(() => {
+    const fetchFriends = async () => {
+      try {
+        const f = await ClientSingleton.getInstance().user.friends.getFriends();
+        setFriends(f);
+      } catch (error) {
+        console.error("Error fetching friends:", error);
+      }
+    };
+
+    fetchFriends();
+    // eslint-disable-next-line
+  }, []);
+
+  /**
+   * On selectedFriend change, fetch and set the user's featuredDeck and the user's featuredDeckCards.
+   */
+  useEffect(() => {
+    if (selectedFriend) {
+      const setDeckAndCards = async () => {
+        try {
+          const featuredD = await selectedFriend.profile.getDeck();
+          setFeaturedDeck(featuredD);
+
+          if (featuredD) {
+            const cards = await featuredD.getCards();
+            setFeaturedDeckCards(cards);
+          }
+        } catch (error) {
+          toast(selectedFriend.username + " has no Featured Deck");
```

Comment:
This toast is in the catch due to an issue with the wrapper where it throws an exception on this line:

```typescript
          const featuredD = await selectedFriend.profile.getDeck();
          ```
          when fetching a null deck